lib/redis/index.ts

```diff
@@ -787,7 +770,14 @@ Redis.prototype.sendCommand = function (command: Command, stream: NetStream) {
         command.args
       );
     }
-    (stream || this.stream).write(command.toWritable());
+
+    if (stream) {
+      if (stream.isPipeline)
+        stream.write(command.toWritable(stream.destination.redis.stream));
```

Comment:
For the pipeline we could move the buffer logic into the Redis class itself.
Maybe even persist the pipeline itself in the queue instead of calling sendCommand that often.